lib/Dialect/AIEVec/Transforms/VectorToAIEVecConversions.cpp

```diff
@@ -2606,6 +2721,48 @@ static void configureAIEVecCommonLegalizations(ConversionTarget &target,
     return false;
   });
 
+  target.addDynamicallyLegalOp<math::CeilOp>([](math::CeilOp ceilOp) {
+    VectorType srcType = dyn_cast<VectorType>(ceilOp.getOperand().getType());
+    if (!srcType) {
+      return true;
+    }
+
+    Type scalarType = srcType.getElementType();
+
+    if (!isa<FloatType>(scalarType)) {
+      return true;
+    }
+
+    unsigned laneSize = getVectorLaneSize(srcType);
+    unsigned elWidth = scalarType.getIntOrFloatBitWidth();
+    if (elWidth != 16 || (laneSize != 16 && laneSize != 32)) {
+      return true;
+    }
+
+    return false;
+  });
+
+  target.addDynamicallyLegalOp<math::FloorOp>([](math::FloorOp floorOp) {
+    VectorType srcType = dyn_cast<VectorType>(floorOp.getOperand().getType());
+    if (!srcType) {
+      return true;
+    }
+
+    Type scalarType = srcType.getElementType();
+
+    if (!isa<FloatType>(scalarType)) {
+      return true;
+    }
+
+    unsigned laneSize = getVectorLaneSize(srcType);
+    unsigned elWidth = scalarType.getIntOrFloatBitWidth();
+    if (elWidth != 16 || (laneSize != 16 && laneSize != 32)) {
```

Comment:
neat: we can just do `return (elWidth != 16 || (laneSize != 16 && laneSize != 32))`.